### DISCUSSION TITLE
Mapping to readonly properties

### DISCUSSION THREAD

**Discussion by mehdihadeli** [LuckyPennySoftware/AutoMapper#3707] [NA]
Hi,
My class is immutable and don't want to set its properties as public, also for implementing immutability I use custom constructor. But I get this error:

> `LoginCommand needs to have a constructor with 0 args or only optional args. (Parameter 'type')`

This is my configurations:

``` csharp
public AuthenticationMapping()
{
    CreateMap<LoginRequest, LoginCommand>();
}

public class LoginRequest
{
    public LoginRequest(string userNameOrEmail, string password, bool rememberMe)
    {
        UserNameOrEmail = userNameOrEmail;
        Password = password;
        RememberMe = rememberMe;
    }

    public string UserNameOrEmail { get; }
    public string Password { get; }
    public bool RememberMe { get; }
}

public class LoginCommand : ICommand<LoginResponse>
{
    public LoginCommand(string userNameOrEmail, string password, bool remember)
    {
        UserNameOrEmail = userNameOrEmail;
        Password = password;
        Remember = remember;
    }

    public string UserNameOrEmail { get; }
    public string Password { get; }
    public bool Remember { get; }
    public Guid Id { get; } = Guid.NewGuid();
}
```
What is the solutions for this purpose? I think one approach is using `.UsingConstructor()` but I have to repeat in my all mapping.

**Comment by lbargaoanu** [comment]
https://docs.automapper.org/en/latest/Construction.html

**Reply by mehdihadeli** [reply]
Thanks for your help.
``` csharp
public class Source {
    public int Value { get; set; }
}
public class SourceDto {
    public SourceDto(int value) {
        _value = value;
    }
    private int _value;
    public int Value {
        get { return _value; }
    }
}
var configuration = new MapperConfiguration(cfg => cfg.CreateMap<Source, SourceDto>());
```
The docs are like what I do, but I got above exception

> LoginCommand needs to have a constructor with 0 args or only optional args. (Parameter 'type')

**Reply by lbargaoanu** [reply]
Try the [MyGet](https://docs.automapper.org/en/latest/The-MyGet-build.html) build.

**Reply by mehdihadeli** [reply]
> Try the [MyGet](https://docs.automapper.org/en/latest/The-MyGet-build.html) build.

After update, I get this error:

> System.TypeLoadException: Could not load type 'AutoMapper.Configuration.MapperConfigurationExpression' from assembly 'AutoMapper, Version=10.0.0.0, Culture=neutral, PublicKeyToken=be96cd2c38ef1005'.

**Reply by lbargaoanu** [reply]
https://github.com/AutoMapper/AutoMapper/issues/3706#issuecomment-918018534
